alacritty/src/config/bindings.rs

```diff
@@ -319,6 +397,34 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
         F19; Action::Esc("\x1b[33~".into());
         F20; Action::Esc("\x1b[34~".into());
         NumpadEnter; Action::Esc("\n".into());
+        Escape, ModifiersState::CTRL;                              Action::ToggleKeyboardMode;
+        Escape,                        +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        I,                             +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        Q,                             +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        V,                             +TermMode::KEYBOARD_MOTION; Action::ToggleNormalSelection;
+        V,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::ToggleLineSelection;
+        V,      ModifiersState::CTRL,  +TermMode::KEYBOARD_MOTION; Action::ToggleBlockSelection;
+        V,      ModifiersState::ALT,   +TermMode::KEYBOARD_MOTION; Action::ToggleSemanticSelection;
```

Comment:
> you should move your cursor appropriately with selection and not behind it, like you're doing it right now.

I'm not sure what you mean by this. Further, why do you think people wouldn't want semantic selection in keyboard motion mode? I personally would like better parity between this mode and the existing mouse selections.

Reply:
because it doesn't make any sense. You have `e` and `b` to contorol it, vim doesn't have such thing either IIRC. This mode only required for mouse, because you don't have control what word is. Also it doesn't work right now the way someone could want.

Reply:
I'm still not following sorry. Calling `ToggleSemanticSelection` while in mouse motion mode works as I was expecting it to. It expands the selection around the full semantic word currently under the cursor.

It would be nice if the movement actions and the semantic selection code was unified however. I'd like a motion for move to exactly the next semantic delimiter.

Reply:
I mean that your semantic selection is just normal selection + you clicking on proper motions(basically `e` and `b` bidning, since they're also semantic aware. So having 2 same things doesn't make any sense.

Reply:
@kchibisov YES! This is what I'm trying to say in https://github.com/alacritty/alacritty/pull/3315#discussion_r377899589.

Reply:
Honestly I just implemented it because it was so easy. Cutting down on bloat that is non-essential sounds like a good idea to me.